### ISSUE TITLE
Multiple failures in the "Creating and deploying automation assets..." section of deploy.ps1.....

### ISSUE THREAD

**Issue by IanMartin56** [OfficeDev/microsoft-teams-apps-requestateam#107]
Hi, deploy.ps1 is 'nearly' working for me but I am getting multiple failures in the 'Creating and deploying automation assets...' section.
All of those errors are below but I believe that they might all originate from the 1st **New-AzAutomationAccount : Operation returned an invalid status code 'BadRequest'** error at line 576. 
I am also getting an error in the earlier **Launching Azure CLI sign-in...** which might be related and is immediately below.
I have tried twice with the same results.
I am using a Dev Tenant + a Trial Subscription.
These are the only errors I am getting.
The (anonymised) command line I am using is: .\deploy.ps1 -TenantName "MyTenant" -TenantID "xxxxxxxx-xxxx-xxx-xxxxxxxxxxx" -RequestsSiteName "Request_a_Team_App_SPOL_Sitev3" -RequestsSiteDesc "V3 SPOL site for the Request a Team App_Used to store Teams Requests" -ManagedPath "sites" -SubscriptionId xxxxxxx-xxxx-xxxx-xxxxxxxxxxxxxxxx -Location "uksouth" -ResourceGroupName "Request_a_Team_App_Resource_Groupv3" -AppName "Request_a_Team_App_AAD_Appv3" -ServiceAccountUPN "Request_Team_App_Svc@MyTenant.onmicrosoft.com" -UseMSGraphBeta $false -IsEdu $false -KeyVaultName "RequestTeamAppKvaultv3"
Can anybody offer any suggestions as to what is going wrong here please?

**Launching Azure CLI sign-in...**
Account                         Environment TenantId                             TenantDomain         AccountType
-------                         ----------- --------                             ------------         -----------
IsaGlobalAdmin AzureCloud  xxxxxxxx-xxxx-xxx-xxxxxxxxxxx MyTenant.onmicrosoft.com User       
az : You have logged in. Now let us find all the subscriptions to which you have access...
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:830 char:13
+ $cliLogin = az login
+             ~~~~~~~~
    + CategoryInfo          : NotSpecified: (You have logged... have access...:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError

**Creating and deploying automation assets...**
Automation account already exists - deleting...
Automation account deleted
Creating automation account...
**New-AzAutomationAccount : Operation returned an invalid status code 'BadRequest'**
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:576 char:9
+         New-AzAutomationAccount -Name $automationAccountName -Locatio ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAutomationAccount], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.NewAzureAutomationAccount

Finished creating automation account
Importing automation modules...
New-AzAutomationModule : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For 
more details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:585 char:9
+         New-AzAutomationModule -AutomationAccountName $automationAcco ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAutomationModule], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.NewAzureAutomationModule

Get-AzAutomationModule : The module was not found. Module name: Az.Accounts.
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:587 char:17
+ ...     while ((Get-AzAutomationModule -Name "Az.Accounts" -ResourceGroup ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzAutomationModule], ResourceNotFoundException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.GetAzureAutomationModule

New-AzAutomationModule : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For 
more details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:591 char:9
+         New-AzAutomationModule -AutomationAccountName $automationAcco ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAutomationModule], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.NewAzureAutomationModule

Finished importing automation modules
Importing and publishing runbooks...
Import-AzAutomationRunbook : The Automation account was not found.
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:598 char:9
+         Import-AzAutomationRunbook -Name "CheckSiteExists" -Path "./r ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Import-AzAutomationRunbook], ResourceNotFoundException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.ImportAzureAutomationRunbook

Publish-AzAutomationRunbook : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto/runbooks/CheckSiteExists' under resource group 
'Request_a_Team_App_Resource_Groupv2' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:603 char:9
+         Publish-AzAutomationRunbook -Name "CheckSiteExists" -Resource ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Publish-AzAutomationRunbook], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.PublishAzureAutomationRunbook

Finished importing and publishing runbooks
Creating automation variables...
New-AzAutomationVariable : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For 
more details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:610 char:9
+         New-AzAutomationVariable -AutomationAccountName $automationAc ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAutomationVariable], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.NewAzureAutomationVariable

New-AzAutomationVariable : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For 
more details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:611 char:9
+         New-AzAutomationVariable -AutomationAccountName $automationAc ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAutomationVariable], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.NewAzureAutomationVariable

Finished creating automation variables
Creating role assignments...
New-AzRoleAssignment : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For more 
details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:618 char:9
+         New-AzRoleAssignment -ObjectId $global:appServicePrincipalId  ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzRoleAssignment], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Resources.NewAzureRoleAssignmentCommand

New-AzRoleAssignment : The Resource 'Microsoft.Automation/automationAccounts/teamsautomate-auto' under resource group 'Request_a_Team_App_Resource_Groupv2' was not found. For more 
details please go to https://aka.ms/ARMResourceNotFoundFix
At C:\SwSetup\microsoft-teams-apps-requestateam-master\microsoft-teams-apps-requestateam-master\Deployment\Scripts\deploy.ps1:619 char:9
+         New-AzRoleAssignment -ObjectId $global:appServicePrincipalId  ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzRoleAssignment], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Resources.NewAzureRoleAssignmentCommand

Finished creating role assignments
Finished automation assets deployment
Deploying api connections...

**Comment by alexc-MSFT** [comment]
@IanMartin56 lets try and get you past initial error first which seems to indicate that the Azure CLI is not installed.

Please can you download and install it from here - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest. Once this is done you will need to close and re-launch PowerShell.

This should solve the 'az login' error where 'az' is a CLI cmdlet.

**Comment by IanMartin56** [comment]
Hi @alexc-MSFT , I have fixed the issue that I had and the app is now working fully.  Azure CLI was installed but not at the latest version so I upgraded it but that did not fix the issue.  The activity log on the Resource Group showed an error re using an account that was already associated with another Resource Group so I removed all previous instances of Resource Groups and then the script completed successfully and as above the app is now working fully.  Thanks for you help.
